src/test/java/com/commercetools/sync/types/utils/typeactionutils/BuildFieldDefinitionUpdateActionsTest.java

```diff
@@ -0,0 +1,419 @@
+    }
+
+    @Test
+    public void buildUpdateActions_WithOneExtraFieldDefinition_ShouldBuildAddFieldDefinitionAction() {
+        final Type oldType = readObjectFromResource(TYPE_WITH_FIELDS_ABC, Type.class);
+
+        final TypeDraft newTypeDraft = readObjectFromResource(
+                TYPE_WITH_FIELDS_ABCD,
+                TypeDraft.class
+        );
+
+        final List<UpdateAction<Type>> updateActions = buildFieldDefinitionUpdateActions(
+                oldType,
+                newTypeDraft,
+                SYNC_OPTIONS
+        );
+
+        assertThat(updateActions).containsExactly(
+            AddFieldDefinition.of(FIELD_DEFINITION_D)
+        );
+    }
+
+    @Test
+    public void buildUpdateActions_WithOneFieldDefinitionSwitch_ShouldBuildRemoveAndAddAFieldDefinitionsActions() {
+        final Type oldType = readObjectFromResource(TYPE_WITH_FIELDS_ABC, Type.class);
+
+        final TypeDraft newTypeDraft = readObjectFromResource(
+                TYPE_WITH_FIELDS_ABD,
+                TypeDraft.class
+        );
+
+        final List<UpdateAction<Type>> updateActions = buildFieldDefinitionUpdateActions(
+                oldType,
+                newTypeDraft,
+                SYNC_OPTIONS
+        );
+
+        assertThat(updateActions).containsExactly(
+            RemoveFieldDefinition.of(FIELD_C),
+            AddFieldDefinition.of(FIELD_DEFINITION_D)
+        );
+    }
+
+    @Test
+    public void buildUpdateActions_WithDifferent_ShouldBuildChangeFieldDefinitionOrderAction() {
+        final Type oldType = readObjectFromResource(TYPE_WITH_FIELDS_ABC, Type.class);
+
+        final TypeDraft newTypeDraft = readObjectFromResource(
+                TYPE_WITH_FIELDS_CAB,
+                TypeDraft.class
+        );
+
+        final List<UpdateAction<Type>> updateActions = buildFieldDefinitionUpdateActions(
+                oldType,
+                newTypeDraft,
+                SYNC_OPTIONS
+        );
+
+
+        assertThat(updateActions).containsExactly(
+            ChangeFieldDefinitionOrder
+                .of(asList(
+                        FIELD_DEFINITION_C.getName(),
+                        FIELD_DEFINITION_A.getName(),
+                        FIELD_DEFINITION_B.getName()
+                ))
+        );
+    }
+
+    @Test
+    public void buildUpdateActions_WithRemovedAndDifferentOrder_ShouldBuildChangeOrderAndRemoveActions() {
+        final Type oldType = readObjectFromResource(TYPE_WITH_FIELDS_ABC, Type.class);
+
+        final TypeDraft newTypeDraft = readObjectFromResource(
+                TYPE_WITH_FIELDS_CB,
+                TypeDraft.class
+        );
+
+        final List<UpdateAction<Type>> updateActions = buildFieldDefinitionUpdateActions(
+                oldType,
+                newTypeDraft,
+                SYNC_OPTIONS
+        );
+
+        assertThat(updateActions).containsExactly(
+            RemoveFieldDefinition.of(FIELD_A),
```

Comment:
Does the order of actions matter? Then your method name has wrong order of methods: `ShouldBuildChangeOrderAndRemoveActions` -> `ShouldBuildRemoveActionsAndChangeOrder`. Also other test methods has wrong names like this.

Reply:
I mean it does not matter in the method names. I changed some of them.

Reply:
In this test you are testing the fact the correct update actions have been built. **Only** in this test, the order doesn't matter. So you should actually assert with `containsExactlyInAnyOrder` not with `containsExactly`.

However, order **definitely** matters which update actions are generated before which. But this should have it's own unit test focusing on testing the order of the generated actions with a strict `containsExactly` assert. This way you don't test two things together.